src/api/tp.js

```diff
@@ -67,19 +68,32 @@ var TargetProcess = (function() {
     }
 
     var me = this;
-    me.getTask(id).then(success, function (err) {
-        if(err.statusCode === 404) {
-            me.getBug(id).then(success, function (err) {
-                if(err.statusCode === 404) {
-                    done('Task/Bug with Id '+id+' could not be found or access is forbidden.');
-                }
-                else failure(err);
-            });
-        }
-        else failure(err);
-    });
+        me.getTask(id).then(success, function (err) {
```

Comment:
these should be changed to chained promises instead of the arrow effect! :)

I know my fault to begin with, but a good chance to fix the styling here.